core/src/main/java/com/sap/ai/sdk/core/AiCoreService.java

```diff
@@ -31,19 +27,21 @@
 
 /** Connectivity convenience methods for AI Core. */
 @Slf4j
-@RequiredArgsConstructor
 public class AiCoreService implements AiCoreDestination {
-
-  Function<AiCoreService, Destination> baseDestinationHandler;
-  final BiFunction<AiCoreService, Destination, ApiClient> clientHandler;
-  final BiFunction<AiCoreService, Destination, DefaultHttpDestination.Builder> builderHandler;
+  static final String AI_CLIENT_TYPE_KEY = "URL.headers.AI-Client-Type";
+  static final String AI_CLIENT_TYPE_VALUE = "AI SDK Java";
+  static final String AI_RESOURCE_GROUP = "URL.headers.AI-Resource-Group";
 
   private static final DeploymentCache DEPLOYMENT_CACHE = new DeploymentCache();
 
-  private static final String AI_RESOURCE_GROUP = "URL.headers.AI-Resource-Group";
+  @Nonnull private final DestinationResolver destinationResolver;
 
-  /** loads the .env file from the root of the project */
-  private static final Dotenv DOTENV = Dotenv.configure().ignoreIfMissing().load();
+  @Nonnull private Function<AiCoreService, HttpDestination> baseDestinationHandler;
```

Comment:
(Question)

Unrelated to your PR I'm wondering why this field is not final.

Reply:
Was wondering that too and planning to create another separate PR after this one with changes